pkg/util/version/version.go

```diff
@@ -181,18 +181,24 @@ func (v *Version) String() string {
 // compareInternal returns -1 if v is less than other, 1 if it is greater than other, or 0
 // if they are equal
 func (v *Version) compareInternal(other *Version) int {
-	for i := range v.components {
+
+	vLen := len(v.components)
+	oLen := len(other.components)
+	for i := 0; i < vLen && i < oLen; i++ {
 		switch {
-		case i >= len(other.components):
-			if v.components[i] != 0 {
-				return 1
-			}
 		case other.components[i] < v.components[i]:
 			return 1
 		case other.components[i] > v.components[i]:
 			return -1
 		}
 	}
+	// If components are common but one has more items, it is bigger
+	switch {
+	case oLen < vLen && (vLen-oLen >= 2 || v.components[oLen] != 0):
+		return 1
```

Comment:
Hm. This fixes "1.2" vs "1.2.0" but not "1.2" vs "1.2.0.0".

Reply:
It's a bit weird use case, but I'll find solution today. I have idea, that I need to verify and will update PR.